packages/gatsby/src/utils/babel-config.js

```diff
@@ -155,6 +155,11 @@ module.exports = async function babelConfig(program, stage) {
     babelrc.plugins.unshift(`react-hot-loader/babel`)
   }
 
+  if (stage !== `develop`) {
+    // Remove prop-types in production.
+    babelrc.plugins.unshift(`transform-react-remove-prop-types`)
```

Comment:
oh! Ummm yes it would :-)

Hmmm ok I naively thought this would be a safe global optimization but clearly not. Thanks for heads up :-) I'll remove it.